conformance/tests/specialtypes_promotions.py

```diff
@@ -0,0 +1,16 @@
+"""
+Tests "type promotions" for float and complex when they appear in annotations.
+"""
+
+# Specification: https://typing.readthedocs.io/en/latest/spec/special-types.html#special-cases-for-float-and-complex
+
+v1: float = 1
+v2: complex = 1.2
+v2 = 1
+
+
+def func1(f: float):
+    f.numerator  # Type error
+
+    if not isinstance(f, float):
+        f.numerator  # OK
```

Comment:
Could be worth having tests for byte + bytearray compatibility, since this was changed by PEP 688

Reply:
This was never part of any spec. It was something that mypy implemented on its own, and I followed suit in pyright but later removed it (as the default behavior). The other type checkers don't implement it (AFAIK). I've added it to the list of topics we should discuss.

Reply:
It was sort of part of the spec, since the CPython docs said that bytearray and memoryview should be treated as compatible with bytes. But of course, PEP 688 explicitly removed the special case. I don't think the spec needs to say anything specific here.